### PR TITLE
add stop time; limit to results, remove redundant logs

### PR DESCRIPTION
**Pull request by gcammarota**

### CODE REVIEW

File: c3s_eqc_automatic_quality_control/dashboard.py [hunk 1]
```diff
@@ -91,47 +93,52 @@ def get_most_recent_log(info: list[dict[Any, Any]]) -> dict[Any, Any]:
     return sorted_info[-1]
 
 
-def update_status_from_logfile(
-    logfile: pathlib.Path, info: dict[Any, Any]
-) -> dict[Any, Any]:
-    with open(logfile, encoding="utf-8") as f:
+def update_from_logfile(logfile: pathlib.Path, info: Dict[Any, Any]) -> Dict[Any, Any]:
```
Comment (malmans2) [inline c3s_eqc_automatic_quality_control/dashboard.py:96]:
```suggestion
def update_from_logfile(logfile: pathlib.Path, info: dict[Any, Any]) -> dict[Any, Any]:
```

File: c3s_eqc_automatic_quality_control/dashboard.py [hunk 2]
```diff
@@ -22,15 +22,17 @@
 import pathlib
 import re
 from operator import itemgetter
-from typing import Any
+from typing import Any, Dict
```
Comment (malmans2) [inline c3s_eqc_automatic_quality_control/dashboard.py:25]:
```suggestion
from typing import Any, dict
```

File: c3s_eqc_automatic_quality_control/dashboard.py (hunk 1 above)
Comment (malmans2) [inline c3s_eqc_automatic_quality_control/dashboard.py:96]:
```suggestion
def update_from_logfile(logfile: pathlib.Path, info: dict[Any, Any]) -> dict[Any, Any]:
```

File: c3s_eqc_automatic_quality_control/dashboard.py [hunk 3]
```diff
@@ -91,47 +93,52 @@ def get_most_recent_log(info: list[dict[Any, Any]]) -> dict[Any, Any]:
...
-    logfile: pathlib.Path, info: dict[Any, Any]
-) -> dict[Any, Any]:
-    with open(logfile, encoding="utf-8") as f:
+def update_from_logfile(logfile: pathlib.Path, info: Dict[Any, Any]) -> Dict[Any, Any]:
+    with open(logfile, "r", encoding="utf-8") as f:
         # get only last matched line
-        for match in map(re.compile(MSG_REGEX).match, reversed(f.readlines())):
-            if match is not None:
-                info.update({"status": match["status"]})
+        lines = f.readlines()
```
Comment (malmans2) [inline c3s_eqc_automatic_quality_control/dashboard.py:99]: Should we avoid reading the whole file at once?
```python
for line in f:
    line = line.strip("\n")
```

File: c3s_eqc_automatic_quality_control/download.py [hunk 4]
```diff
@@ -278,12 +278,11 @@ def split_request(
 def download_and_transform_chunk(
     collection_id: str,
     request: dict[str, Any],
-    func: None
+    transform_func: None
```
Comment (malmans2) [inline c3s_eqc_automatic_quality_control/download.py:281]: OK with me. Why though?
If the issue is cacholote, I've fixed it a couple of days ago.

File: c3s_eqc_automatic_quality_control/runner.py [hunk 5]
```diff
@@ -22,7 +22,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 from inspect import getmembers, isfunction
-from typing import Any
+from typing import Any, Tuple
```
Comment (malmans2) [inline c3s_eqc_automatic_quality_control/runner.py:25]:
```suggestion
from typing import Any
```

File: c3s_eqc_automatic_quality_control/runner.py [hunk 6]
```diff
@@ -64,26 +64,29 @@ def show_config_template() -> None:
...
+    return [
+        f[0] for f in getmembers(diagnostics, isfunction) if not f[0].startswith("_")
+    ]
 
 
 def process_request(
     request: dict[Any, Any],
     logger: logging.Logger = LOGGER,
-) -> dict[Any, Any]:
+) -> Tuple[dict[Any, Any], list[str]]:
```
Comment (malmans2) [inline c3s_eqc_automatic_quality_control/runner.py:75]:
```suggestion
) -> tuple[dict[Any, Any], list[str]]:
```

**Review by malmans2 — APPROVED** [review]

File: c3s_eqc_automatic_quality_control/download.py (hunk 4 above)
Comment (gcammarota) [reply c3s_eqc_automatic_quality_control/download.py:281]: It raises a `TypeError: hexdigestify_python_call() got multiple values for argument 'func'`. Good to know it's fixed on cacholote side. On top of that `transform_func` could be a clearer name. Explicit it's better than implicit they say, right?

Comment (malmans2) [reply c3s_eqc_automatic_quality_control/download.py:281]: sounds good, I'll update the notebooks then

File: c3s_eqc_automatic_quality_control/cli.py [hunk 7]
```diff
@@ -69,14 +66,19 @@ def run(
 def dasboard(
```
Comment (malmans2) [inline c3s_eqc_automatic_quality_control/cli.py:66]:
```suggestion
def dashboard(
```

File: c3s_eqc_automatic_quality_control/dashboard.py [hunk 8]
```diff
@@ -63,10 +67,14 @@ def set_logfile(logger: logging.Logger, logfilepath: pathlib.Path) -> logging.Lo
...
-        log_dir.mkdir(parents=True, exist_ok=True)
-    return log_dir
+    if os.environ.get(EQC_AQC_ENV_VARNAME) is None:
+        eqc_dir = pathlib.Path.home() / ".eqc"
+    else:
+        eqc_dir = pathlib.Path(os.environ[EQC_AQC_ENV_VARNAME])
+    log_dir_path = eqc_dir / "logs/"
+    if not log_dir_path.is_dir():
+        log_dir_path.mkdir(parents=True, exist_ok=True)
+    return log_dir_path
```
Comment (malmans2) [inline c3s_eqc_automatic_quality_control/dashboard.py:77]: I think you can get rid of the if statements as pathlib takes care of it:
```suggestion
    eqc_dir = os.environ.get(EQC_AQC_ENV_VARNAME, pathlib.Path.home() / ".eqc")
    log_dir_path = pathlib.Path(eqc_dir) / "logs"
    log_dir_path.mkdir(parents=True, exist_ok=True)
    return log_dir_path
```

**Review by malmans2 — APPROVED** [review]

**Pull request merged**